Dev1

Spartwo opened the pull request.

Spartwo committed:
Misc

- Allowed for TURD on fireworks while restock(which does not modify the fireworks) is installed
- Updated KALTrigger description
- Simplified decoupler designation
- Re-added deleted performance saving AI timestep setting
- Withdrawal allowance setting

Spartwo committed:
Merge branch 'dev' of https://github.com/Halbann/StockCombatAI into dev

Spartwo commented:
- Allowed for TURD on fireworks while restock(which does not modify the fireworks) is installed
- Updated KALTrigger description
- Simplified decoupler designation
- Re-added deleted performance saving AI timestep setting
- Withdrawal allowance setting

Spartwo committed:
Merge branch 'dev' of https://github.com/Halbann/StockCombatAI into dev

Halbann commented on GameData/KCS/Patches/FireworkLauncher.cfg:
The texture set 'Stock_Default_fireworksLaunchers' doesn't exist when Restock is installed so I don't need to patch it. For this to make sense you ought to include another patch that replicates the TU configs for the firework launcher that don't have the !Restock condition, otherwise leave it as is.

Spartwo replied:
It doesn't exist when restock is installed because tu_stockrecolour includes the same module manager block on all parts. Having it is creating the need for a large block of "no restock allowed" compatibility patches for KCS in any circumstance where TURD and ReStock are being used concurrently.

Halbann commented on Source/KerbalCombatSystems/CombatRobotics.cs:
This comment seems out of context?

Halbann commented on Source/KerbalCombatSystems/ObjectTracking.cs:
It's not worthwhile for you to change my formatting like this. I'm going to continue writing multi-line attributes like this so please undo.

Spartwo replied:
It's not worthwhile for you to change my formatting like this. I'm going to continue to indent text like this so please accept.

Halbann replied:
I’m sorry man but Im not interested, this is just making me really miserable. I don’t want to give my free time to work with you

Halbann commented on Source/KerbalCombatSystems/Settings.cs:
The time between behaviour updates is not designed as a performance setting. If it was I would make it a setting. Please remove.

Halbann replied:
To explain, hopefully conclusively, the ship controller is designed to run at a nominally fixed update rate, for which I have determined the maximum feasible time to be about 3 seconds. Running at a known maximum rate allows me to make optimisation decisions about what kind of computation to put in the behaviour tree and what to put in each individual behaviour. The design of the tree and the behaviours follows from that decision. The AI **will not** function properly at a higher interval because it is not designed to.

I'm not saying that the design is ideal or that it's implemented in an ideal way, only that this is how it functions.

I appreciate that you're looking out for performance but your understanding of how the ship controller works is clearly wrong. It's also wrong to make unfounded decisions like this when we have profiling available to us, through which you've already seen me identify and optimise a bottleneck (CheckStatus). No one has identified any performance problems relating to the ship controller besides that.

Lastly, if you're going to repeatedly force through a change like this, you need to argue why in-detail, otherwise you're making the decision on your own. It's totally unreasonable to think that you don't need any input from me when making such a huge change to the thing that I designed.

Halbann commented on Source/KerbalCombatSystems/DecouplerDesignator.cs:
Please could you undo the changes you made to the attribute formatting here

Halbann commented on Source/KerbalCombatSystems/EscapePod.cs:
Please remove the implementation of a refresh rate setting.

Spartwo committed:
capitalisation

Pascal => Camel case in robotics caused compile error

Spartwo commented on GameData/KCS/Patches/FireworkLauncher.cfg:
Though (as seen in discord) I am working on a substitude block of module manager patches for this case it's a lot more trouble than the simple removal of that requirement under the assumption that anyone using restock is using the compatible texture swapper as is the case with myself

Spartwo commented:
Well hopefully some performance can be saved elsewhere if it ever comes down to it. I unremoved it after those performance tests showing those occasional updates being the by far the most demanding functions that KCS runs.

Spartwo committed:
Disabled refresh rate

as requested

Spartwo committed:
Re-place restock block

will add required edge case specifics once modules are available in the repo to test with
